base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -325,8 +325,29 @@ impl AggregateBody {
         Ok(())
     }
 
-    /// This function will check all stxo to ensure that feature flags where followed
-    pub fn check_stxo_rules(&self, height: u64) -> Result<(), TransactionError> {
+    pub fn check_output_features(&self, max_coinbase_metadata_size: usize) -> Result<(), TransactionError> {
+        for output in self.outputs() {
```

Comment:
nit(perf): Rather than looping each time in the validation helper, reuse the loop and call this with the output.

![image](https://user-images.githubusercontent.com/1057902/203963180-37d8c454-c7f2-4b10-975f-ee24816448b7.png)